### ISSUE TITLE
A question about BEIT pre-training loss

### ISSUE THREAD

**Issue by cmh2019** [microsoft/unilm#388]
Hello,
Thank you for your great work BEIT. I have a question about the loss. In the section 2.3 of BEIT paper, you say the pre-training objective is definded in formula 1. In formula 1, it seems the loss is calculated only by the masked tokens. https://github.com/microsoft/unilm/blob/b58d3ca6da7b8521b0f7bee7d9fcf6741d82233f/beit/engine_for_pretraining.py#L55 However, it seems this code use all the tokens (not only the masked tokens) to calculate the loss. I wonder which one is right or whether the code is slightly different with the paper? Thanks in advance!

**Comment by addf400** [comment]
@cmh2019 , thanks for your kind words. 
`bool_masked_pos` is a bool tensor that indicates whether a token is masked or not. 
The code uses `bool_masked_pos` to select the masked visual tokens/features as follows:
https://github.com/microsoft/unilm/blob/b58d3ca6da7b8521b0f7bee7d9fcf6741d82233f/beit/engine_for_pretraining.py#L52
And the model will return the features of masked patches by:
https://github.com/microsoft/unilm/blob/b58d3ca6da7b8521b0f7bee7d9fcf6741d82233f/beit/modeling_pretrain.py#L135

**Comment by cmh2019** [comment]
Thanks a lot!

**Comment by donglixp** [comment]
The code and pre-trained models of BEiT-3 can be found at [aka.ms/beit3](https://aka.ms/beit3).
